### ISSUE TITLE
[Bug]: init_hook never running

### ISSUE THREAD

**Issue by gfriloux** [jetpack-io/devbox#652]
**Current Behavior (bug)**
Setting an `init_hook` has no effect

**Expected Behavior (fix)**
It should run when using `devbox shell` or `devbox run script_name`

**Additional context**
Following https://www.jetpack.io/devbox/docs/guides/scripts/, i add a `script` section to my file:
```json
{
  "packages": [
    "just",
    "shellcheck",
    "jq",
    "bc",
    "awscli2"
  ],
  "shell": {
    "init_hook": "echo \"Hello \"",
    "scripts": {
        "echo_once": "echo \"World\"", 
        "echo_twice": [
            "echo \"World\"",
            "echo \"Again\""
        ]
    }
  },
  "nixpkgs": {
    "commit": "52e3e80afff4b16ccb7c52e9f0f5220552f03d04"
  }
}
```

I run `devbox run echo_once`, that should display `Hello\nWorld`, but get this result:
```
Ensuring packages are installed.
Starting a devbox shell...
World
```

Tools used:
* `I use arch BTW`
* `Fish 3.6.0` (same issue using `bash` from `fish`)
* `devbox 0.3.3`

**Comment by gfriloux** [comment]
Spent time to try if this was due to a regression.
Went back to version 0.2.2 (which introduce `devbox run`).

So i can tell that it just never worked in my setup.

It seems related to https://github.com/jetpack-io/devbox/issues/401 that fix things for `fish`, but it may be incomplete.

**Comment by ipince** [comment]
Hi Guillaume. Thanks for the report! I'll take a look later today.

**Comment by ipince** [comment]
OK I was able to reproduce and will investigate a fix.

**Comment by ipince** [comment]
This should be fixed in version `0.4.0`
